test/FastTests/Graph/BasicGraphQueries.cs

```diff
@@ -444,6 +444,18 @@ public void CanHandleFiltersInMultiHopQueryWithEndNode()
             }
         }
 
+        [Fact]
+        public void Query_with_non_existing_collection_should_fail()
+        {
+            using (var store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    Assert.Throws<RavenException>(() => session.Advanced.RawQuery<JObject>(@"match (FooBar)").ToList());
```

Comment:
Shouldn we check exception message to ensure we got the exception that we really expected?

Reply:
BTW this assertion fails :)
`Expected: typeof(Raven.Client.Exceptions.RavenException)\r\nActual:   typeof(Raven.Client.Exceptions.InvalidQueryException): `